Cases: Update multiple cases - Change a role in many cases

khennhofer opened the pull request:
Could it be possible to have a chance to include "case roles" into a profile for changing a role for multiple cases?

totten committed:
(REF) Extensions - Extract method to build upgrade queue

I was slightly concerned that building a queue under a different there are
two risks, although I don't think either is very big:

1.  That someone references the queue by name during the building process.
    However, it wouldn't make much sense from a DX POV (given how "Upgrader"
    classes are generally written), and I can't any references to
    `QUEUE_NAME` or `ext-upgrade` in `universe`

2.  That someone generates queue tasks with `weight`s that somehow clash with
    the `weight`s in core-queue. However, this also doesn't make much sense
    from a DX POV (given how "upgrader" classes are generally written).

totten committed:
(REF) Extract method \CRM_Upgrade_DispatchPolicy::useTemporarily()

totten committed:
Upgrader - Enqueue extension upgrade tasks

totten committed:
Add release-notes/5.50.1.md

totten committed:
Add release-notes/5.50.2.md

tschuettler committed:
Fix typo in version of Guzzle

totten committed:
Add release-notes/5.50.3.md

seamuslee001 committed:
[REF][PHP8.1] Fix a couple of deprecations in php8.1 by specifying that return type may change in BaseParamQuery and by ensuring that we don't pass null into function_exists in I18n class

Replace some ReturnTypeWillChange with later return types as they should be backward compatible

Dave D Fix

Completely switch to just not empty on the customTranslationFunction check

mlutfy committed:
Standalone: add setup plugin so that we can install

totten committed:
Add release-notes/5.50.4.md

eileenmcnaughton committed:
Add second currency example

eileenmcnaughton committed:
Fix Parser classes to use ClassScanner

colemanw committed:
Afform - Switch to APIv4 for managed afform_type option group

eileenmcnaughton committed:
Merge pull request #23898 from civicrm/5.51

5.51 to master

totten committed:
(dev/core#3660) CRM_Extension_ClassLoader - Defend against redundant refreshes

Overview
--------

This is a follow-up to #23824 (c24dd7db7e1e91120fd7daeb7e151f856d6b78c3) which addresses a
regressive edge-case.

Steps to Reproduce
------------------

* Write an extension like `wmf-civicrm` which (a) calls `System.flush` (`rebuildMenuAndCaches()`)
  during `hook_install` -- and then (b) loads some class from the same extension.
    ```php
    function foo_civicrm_install() {
      civicrm_api3('System', 'flush', []);
      CRM_Foo_Helper::doStuff();
    }
    ```
* Try to install the extension.

Before
------

Crashes on loading the class `CRM_Foo_Helper`

After
-----

Loads the class `CRM_Foo_Helper`.

Comments
--------

(1) To see what's happening, consider `CRM_Extension_Manager_Module::onPreInstall()`.
This registers the new classloader and then fires `hook_install` which eventually
fires `rebuildMenuAndCaches()`. With c24dd7db, this resets the classloader again.
But the extension isn't fully installed yet - so it forgets about the new extension.

(2) Is it safe to have some (temporarily) sticky items?  Ish.  You might
say: "Ah, but what if we need to remove an extension?  Won't this static
variable retain stale things?" Doesn't matter.  In PHP, classloading is a
one-way-street.  (You cannot unload.) So you'll still have old classes in
memory, regardless of whether the `ClassLoader` has some old metadata
about where to find classes.

(3) I'm on the fence about whether this patch is a good idea.  Calling
`System.flush` explicitly in this context seems like an invitation to
trouble.  OTOH, it worked before #23824, so it can be called a regression.

MegaphoneJon committed:
Grants appear in afform

totten committed:
(NFC) Expand docblocks for hook_install and hook_postInstall

totten committed:
Merge pull request #23891 from eileenmcnaughton/smarty

Reset smarty-security after an exception

totten committed:
Examples - Title should differentiate EUR/CAD examples

totten committed:
Merge pull request #23894 from eileenmcnaughton/off_it

Minor test cleanup

eileenmcnaughton committed:
Merge pull request #23900 from totten/5.51-sticky-classloader

(dev/core#3660) CRM_Extension_ClassLoader - Defend against redundant refreshes

eileenmcnaughton committed:
Allow tests to pick up hook too

eileenmcnaughton committed:
Merge pull request #23901 from totten/nfc-install-docblock

(NFC) Expand docblocks for hook_install and hook_postInstall

eileenmcnaughton committed:
Merge pull request #23889 from eileenmcnaughton/offline_cur

Add second currency example

eileenmcnaughton committed:
dev/core#3503, dev/core#3492 Fix grant exports and relative dates

This both stem from the wrong fix for https://lab.civicrm.org/dev/core/-/issues/3069

Refer to the gitlab for some famous last words....

eileenmcnaughton committed:
Merge pull request #23902 from civicrm/5.51

5.51 to master

eileenmcnaughton committed:
Merge pull request #23903 from eileenmcnaughton/551

Allow tests to pick up hook too

totten committed:
Import Summary - Update to match final schema ('UserJob.job_type')

eg When running "Import Activities", the page-title for the summary is incorrect.

eileenmcnaughton committed:
Merge pull request #23888 from eileenmcnaughton/user_import

Convert the type on the UserJob entity to be a string

eileenmcnaughton committed:
Merge branch '5.51' of github.com:civicrm/civicrm-core into up

eileenmcnaughton committed:
Merge pull request #23907 from eileenmcnaughton/up

5.51 to master

eileenmcnaughton committed:
Merge pull request #23893 from eileenmcnaughton/user_two

dev/core#3514 Add interface to allow extensions that deal with imports to cope with 5.51

eileenmcnaughton committed:
Merge pull request #23909 from civicrm/5.51

5.51 to master

eileenmcnaughton committed:
Merge pull request #23871 from agileware/CIVICRM-2004

Personal Campaign Pages (PCP) incorrectly displays "Don't list my contribution in the honour roll"

mlutfy committed:
Standalone: Tweak css/menubar-standalone.css for the page title and mobile support

mlutfy committed:
CRM_Report_Form: set _from and _where as public for the alterReportVar hook

colemanw committed:
Merge pull request #23899 from MegaphoneJon/civigrant-afform

Grants appear in afform

seamuslee001 committed:
[REF][PHP8.1] Upgrade Symfony to 4.x for WP, Drupal7, Backdrop, Joomla

seamuslee001 committed:
[REF][PHP8.1] Ensure that NULL is not passed to substr in CRM_Utils_File::addTrailingSlash

agileware-fj committed:
CIVICRM-1861 Embed timezone output for iCal
    CIVICRM-1988 Add HTML alternate description to iCAL
    CIVICRM-1987 Add to Google Calendar Link

totten committed:
WorkflowMessageTest - Fix weird inconsistency in resolving `@dataProvider`

We see a failure scenario where:

* The first execution of WorkflowMessageTest runs fine.
* The second execution of WorkflowMessageTest fails to boot (while evaluating `@dataProvider`)
* If you reset the DB, then it works again - but only once.

eileenmcnaughton committed:
Merge pull request #23916 from totten/master-wfmsg-data-boot

WorkflowMessageTest - Fix weird inconsistency in resolving `@dataProvider`

demeritcowboy committed:
Merge pull request #23914 from seamuslee001/file_substr_null

[REF][PHP8.1] Ensure that NULL is not passed to substr in CRM_Utils_F…

colemanw committed:
CrmUi - Add crmUiIconPicker Angular widget

colemanw committed:
Afform - Allow picking icon

Allows afforms placed as a contact summary tab to specify an icon.
It also switches Afforms's hook_civicrm_tabset to use the API, which ensures virtual forms get picked up.
Adds the new icon for CiviGrant as an example.

MegaphoneJon committed:
fixes #3706 URL validation regression

eileenmcnaughton committed:
Merge pull request #23920 from MegaphoneJon/url-validation-regression

fixes #3706 URL validation regression

seamuslee001 committed:
[REF][PHP8.1] Stop Passing NULL values into mb_strlen in DAO nad strpos in API validate string function

seamuslee001 committed:
Merge pull request #23921 from civicrm/5.51

5.51 to master

colemanw committed:
Angular - Support popping up an afform or other ang module via ajax modal

Fixes the AngularLoader to work with modal dialogs via ajax.
Dedupes the already-loaded modules to only load what's needed.

colemanw committed:
crmUi - Fix crmPageTitle directive to work inside a modal

seamuslee001 committed:
[REF][PHP8.1] Upgrade Pear/mail_mime package to support php8.1

seamuslee001 committed:
[REF][PHP8.1] Add in type hints to fix deprecations and add in #[\ReturnTypeWillChange] where type hint would be transversable or mixed for php7 compatability

seamuslee001 committed:
[REF][PHP8.1] Fix default values which triggeres deprecation notices in php8.1

Switch to non array::value format

eileenmcnaughton committed:
dev/core#3707 Fix wordpress issue of page not refreshing after import

artfulrobot committed:
Initial go at cleaning up repeattransaction

demeritcowboy committed:
Merge pull request #23923 from seamuslee001/upgrade_mail_mime

[REF][PHP8.1] Upgrade Pear/mail_mime package to support php8.1

eileenmcnaughton committed:
Upgrade script for contribution_recur amount, fix loading

alifrumin committed:
[NFC] 5.51 release notes

eileenmcnaughton committed:
RepeatTransaction - separate out repeat pledge handling from complete

demeritcowboy committed:
Merge pull request #23926 from seamuslee001/fix_default_value_php81

[REF][PHP8.1] Fix default values which triggeres deprecation notices …

demeritcowboy committed:
add comment

eileenmcnaughton committed:
Merge pull request #23930 from demeritcowboy/trigger-comment

[NFC] Add code comment about triggerinfo "variables"

webmaster-cses-org-uk committed:
Add support for multiple-selection of contact IDs

Update to allow for multiple contact IDs (array) in _civicrm_api3_custom_data_get().

eileenmcnaughton committed:
Import fix - if there are no required fields validateRequiredFields should 'pass'

eileenmcnaughton committed:
dev/core#3713 Use front-end urls for event url tokens

webmaster-cses-org-uk committed:
Correct indentation

Change indentation from 4 spaces to 2, per standards.

seamuslee001 committed:
[REF][PHP8.1] Another set of fixes for passing in NULL values to php string or preg replace functions

Update to Dave's suggestion for default value

mattwire committed:
Merge pull request #23808 from agileware/CIVICRM-1861

issues/2887 Enhancements to improve the iCal, Outlook and Google Calendar integration

demeritcowboy committed:
Make the activity search filter on manage case less unwieldy

seamuslee001 committed:
Merge pull request #23922 from seamuslee001/str_func_null_1

[REF][PHP8.1] Stop Passing NULL values into mb_strlen in DAO nad strp…

demeritcowboy committed:
Merge pull request #23919 from colemanw/afformIcon

Afform - Allow picking icon for tab, add CrmUiIconPicker widget

yashodha committed:
Merge pull request #23912 from mlutfy/reportVars

CRM_Report_Form: set _from and _where as public for the alterReportVar hook

demeritcowboy committed:
Merge pull request #23913 from seamuslee001/symfony_upgrade

[REF][PHP8.1] Upgrade Symfony to 4.x for WP, Drupal7, Backdrop, Joomla

artfulrobot committed:
Fix broken tests highlighted by but not caused by pull 23928

seamuslee001 committed:
[REF][PHP8.1] Update Brick/Math to a version that works with php8.1

seamuslee001 committed:
Merge pull request #23904 from eileenmcnaughton/grant

dev/core#3503, dev/core#3492 Fix grant exports and relative dates

eileenmcnaughton committed:
Merge pull request #23938 from seamuslee001/brick_math_update

[REF][PHP8.1] Update Brick/Math to a version that works with php8.1

eileenmcnaughton committed:
Merge pull request #23939 from civicrm/5.51

5.51

tschuettler committed:
Fix sorting by mapping type

artfulrobot committed:
Add docblock comments to explain $input param in CRM_Contribute_BAO::repeatTransaction

artfulrobot committed:
Minor tidyup of api3 completetransaction; plus comments

mattwire committed:
Merge pull request #23928 from artfulrobot/artfulrobot-cleanup-repeattransaction

Cleanup repeattransaction

demeritcowboy committed:
Merge pull request #23935 from seamuslee001/fix_more_null

[REF][PHP8.1] Another set of fixes for passing in NULL values to php …

demeritcowboy committed:
Merge pull request #23927 from eileenmcnaughton/url

dev/core#3707 Fix wordpress issue of page not refreshing after import

demeritcowboy committed:
Merge pull request #23943 from civicrm/5.51

5.51

mattwire committed:
Merge pull request #23931 from eileenmcnaughton/repeat_pledge

RepeatTransaction - separate out repeat pledge handling from completeOrder

eileenmcnaughton committed:
Merge pull request #23942 from tschuettler/3717-sort-mapping-page

dev/core#3717 - Fix sorting by mapping type

eileenmcnaughton committed:
Enotice fix

seamuslee001 committed:
[REF] Upgrade DOMPDF to 2.x branch

seamuslee001 committed:
[REF] Upgrade DOMPDF to 2.x branch

totten committed:
CRM_Utils_Array - All `pathGet()`-like methods will accept empty-path

These two are equivalent:

  (A) $data['sub1']['sub2']['sub3']
  (B) CRM_Utils_Array::pathGet($data, ['sub1', 'sub2', 'sub3'])

But what if the list of subpaths is empty -- ie you're accessing the top-level item?

  (A) $data
  (B) CRM_Utils_Array::pathGet($data, [])

It worked correctly on `pathGet()`, but accessing the top-level using a
related methods (eg `pathSet()`) would generate a PHP error.

This commit ensures that related methods can be used to check/update the
top-level item (ie path===[]).

Note that there is some decent test-coverage already for the regular
scenarios (eg `testGetSetPathParts()`).

totten committed:
dev/core#3181 - Resolver - Compatibility fix for GLOBALS in PHP 8.1

seamuslee001 committed:
Fix installed-fonts cache

seamuslee001 committed:
Fix installed-fonts cache

eileenmcnaughton committed:
Merge pull request #23949 from alifrumin/rn5.51

[NFC] 5.51 Release Notes - first pass

eileenmcnaughton committed:
Merge pull request #23951 from civicrm/5.51

5.51 to master

eileenmcnaughton committed:
dev/core#3719 fix inconistent handling of job_type:label

eileenmcnaughton committed:
Poly-fill-me-up

mattwire committed:
Merge pull request #23346 from mattwire/totaltaxamount

Fix calculation and assignment of taxAmount on contribution page confirmation

yashodha committed:
Merge pull request #23945 from eileenmcnaughton/map

Enotice fix

agh1 committed:
5.51.0 release notes: added late changes

totten committed:
Merge pull request #23947 from totten/master-php81-global

dev/core#3181 - Resolver - Compatibility fix for GLOBALS in PHP 8.1

colemanw committed:
Afform - Use urlHash embedded in popup dialog

colemanw committed:
Afform - Close popup dialog after submitting

colemanw committed:
Merge pull request #23953 from eileenmcnaughton/imp_job

dev/core#3719 fix inconistent handling of job_type:label

colemanw committed:
Merge pull request #23934 from eileenmcnaughton/url_tokens

dev/core#3713 Use front-end urls for event url tokens

colemanw committed:
Merge pull request #23933 from eileenmcnaughton/require

Import fix - if there are no required fields validateRequiredFields should 'pass'

eileenmcnaughton committed:
Enotice fix for when campaign is disabled

seamuslee001 committed:
Merge pull request #23956 from agh1/5.51.0-releasenotes-final

5.51.0 release notes final changes

eileenmcnaughton committed:
Fix handling of parameters in repeattransaction

The function was forcing in the value from the original contribution
into amount, rather than the value from the templateContribution

eileenmcnaughton committed:
Declare total_amount, make amount an alias

The confusion between amount & total_amount is in
multiple places - this means they are treated 'the same' if passed
into repeattransaction

totten committed:
case-xml@1 - Add example+assertions of new activity-type

totten committed:
mgd-php@1 - Add example+assertions for new case-type (but without a real `definition`)

seamuslee001 committed:
Merge pull request #23958 from civicrm/5.51

5.51

seamuslee001 committed:
[REF][PHP8.1] Fix deprecations where by calling trait static functions when not from a class that implements the trait is deprecated

eileenmcnaughton committed:
Merge pull request #23084 from pradpnayak/paypalipnchangemaount

Subscription Amount change not respected in Paypal

eileenmcnaughton committed:
Remove another instance of pledges using contribution option group

eileenmcnaughton committed:
Merge pull request #23959 from totten/master-casexml-acttype

(NFC) case-xml@1 - Add example+assertions of new activity-type

seamuslee001 committed:
Merge pull request #23960 from seamuslee001/fix_static_calling_trait

[REF][PHP8.1] Fix deprecations where by calling trait static function…

seamuslee001 committed:
Merge pull request #23961 from totten/master-mgdphp-caseacttype

(NFC) mgd-php@1 - Add example+assertions for new case-type

eileenmcnaughton committed:
Smarty notice fix

eileenmcnaughton committed:
Do not escape error message

eileenmcnaughton committed:
Add precautionary purify

eileenmcnaughton committed:
Also don't escape in if context

- just a bit less confusing when debugging

mattwire committed:
Merge pull request #23924 from colemanw/afformPopup

Angular - Support popping up an afform or other ang module via ajax modal

mattwire committed:
Merge pull request #23963 from eileenmcnaughton/tpl

Smarty notice fix

mattwire committed:
Merge pull request #23962 from eileenmcnaughton/pledge

Remove another instance of pledges using contribution option group

mattwire committed:
Merge pull request #23937 from demeritcowboy/case-activity-search-filter

dev/core#3709 - Make the activity search filter on manage case less unwieldy

colemanw committed:
Merge pull request #23964 from eileenmcnaughton/tpl_err

Do not escape error message

colemanw committed:
Merge pull request #23809 from eileenmcnaughton/invoice_recur

Fix bug whereby editing contribution_recur.amount was not updating single-line-item-template-contributions

colemanw committed:
Merge pull request #23929 from eileenmcnaughton/recur

Upgrade script for contribution_recur amount, fix loading

MegaphoneJon committed:
Fixes core#3725 'Access CiviContribute' can't view a single contribution

totten committed:
Merge pull request #23925 from seamuslee001/type_hints_2

[REF][PHP8.1] Add in type hints to fix deprecations and add in #[\Ret…

eileenmcnaughton committed:
Merge pull request #23965 from MegaphoneJon/contribution-view-perm-fix

Fixes core#3725 'Access CiviContribute' can't view a single contribution

colemanw committed:
REF - Remove unneeded return from private fn, add comments

eileenmcnaughton committed:
Fix import falling over when a note is included

MegaphoneJon committed:
Test Only - show FALSE is stored as NULL on custom boolean fields

eileenmcnaughton committed:
Merge pull request #23967 from colemanw/angularCleanup

REF - Remove unneeded return from private fn, add comments

seamuslee001 committed:
Add in necessary patch for html2text for php8.1

seamuslee001 committed:
[REF] dev/core#3676 Add in html2text/html2text to replace usage of roundcube's version we have in packages

colemanw committed:
APIv4 - Fix mishandling of boolean custom values

colemanw committed:
Afform - Allow groups and tags and age to be filters

seamuslee001 committed:
Merge pull request #23968 from eileenmcnaughton/import_note

Fix import falling over when a note is included

seamuslee001 committed:
Update Base Mailing System Test to match newer output format

seamuslee001 committed:
Merge pull request #23971 from seamuslee001/lab_core_3676

[REF] dev/core#3676 Add in html2text/html2text to replace usage of roundcube's version we have in packages

totten committed:
(NFC) SoapTest - Skip this test on Drupal8+ and Backdrop

Never passed on D8+.  Has been failing for a while on Backdrop.  There's a
work-a-like extension.  No need to monitor results on these configurations.

Commit:
Set version to 5.52.beta1

Commit:
Set version to 5.53.alpha1

seamuslee001 committed:
Merge pull request #23975 from totten/5.52-skip-soaptest

(NFC) SoapTest - Skip this test on Drupal8+ and Backdrop

seamuslee001 committed:
Merge pull request #23976 from civicrm/5.52

5.52

demeritcowboy committed:
Merge pull request #23851 from seamuslee001/php81_deprecations

[REF][PHP8.1] Fix a couple of deprecations in php8.1 by specifying th…

demeritcowboy committed:
Merge pull request #23946 from seamuslee001/dompdf_upgrade

[REF] Upgrade DOMPDF to 2.x branch

demeritcowboy committed:
whitespace fix

demeritcowboy committed:
whitespace fix

demeritcowboy committed:
Merge pull request #22918 from demeritcowboy/guzzle7

dev/drupal#171 - Allow guzzle 7 for drupal 10

demeritcowboy committed:
fix lock hash

demeritcowboy committed:
Merge pull request #20588 from mattwire/recaptcharefactor

REF dev/core#2571 Move reCAPTCHA code to extension

eileenmcnaughton committed:
Merge pull request #23977 from demeritcowboy/dompdf-whitespace

[NFC] dompdf patch whitespace fix

eileenmcnaughton committed:
Merge pull request #23978 from demeritcowboy/composerlock

[NFC] fix composer.lock hash

eileenmcnaughton committed:
Comment fixes

eileenmcnaughton committed:
Remove call to deprecated function

eileenmcnaughton committed:
Remove hard coded activity_type_id

eileenmcnaughton committed:
Merge pull request #23932 from webmaster-cses-org-uk/patch-4

Add support for multiple-selection of contact IDs in _civicrm_api3_custom_data_get()

seamuslee001 committed:
[REF] Upgrade oauth2-client,zetacomponents/base,htmlpurfier packages to versions that support php8.1

seamuslee001 committed:
Merge pull request #23979 from eileenmcnaughton/act_status

dev/core#3692 Comment fixes

seamuslee001 committed:
Merge pull request #23981 from eileenmcnaughton/act_seq

dev/core#3692 Remove call to deprecated function

seamuslee001 committed:
Merge pull request #23982 from eileenmcnaughton/act_cont

dev/core#3692 Remove hard coded activity_type_id

seamuslee001 committed:
[NFC] Convert Deprecations in tests to exceptions

seamuslee001 committed:
[NFC] Convert Deprecations in tests to exceptions

seamuslee001 committed:
Merge pull request #23985 from seamuslee001/nfc_deprecations

[NFC] Convert Deprecations in tests to exceptions

seamuslee001 committed:
[NFC] Add in convertDeprecationsToExceptions='true' to core extension's phpunit.xml.dist files

seamuslee001 committed:
[NFC] Add in convertDeprecationsToExceptions='true' to core extension's phpunit.xml.dist files

demeritcowboy committed:
Merge pull request #23986 from seamuslee001/dep_exceptions_ext

[NFC] Add in convertDeprecationsToExceptions='true' to core extension…

demeritcowboy committed:
Merge pull request #22227 from mlutfy/standalone

CiviCRM Standalone: base classes

totten committed:
Civi\Test - Fix leak in active module-list for headless test

Steps to reproduce
------------------

Observed on a `dmaster` build while running these steps:

* Run civix's `make-example.sh`
* Or just run some subset involving a headless test, eg
   * civix generate:test --template=headless 'Civi\Civiexample\BarTest'
   * cv api extension.install key=$EXMODULE
   * phpunit8 ./tests/phpunit/Civi/Civiexample/BarTest.php

Before
------

Fatal error: Uncaught Error: Class 'Civi\Api4\SearchSegment' not found in
/home/totten/bknix/build/dmaster/web/sites/all/modules/civicrm/ext/search_kit/Civi/Api4/Service/Spec/Provider/SearchSegmentExtraFieldProvider.php
on line 52

After
-----

Test runs

seamuslee001 committed:
Merge pull request #23988 from seamuslee001/5_52_phpunit_deprecations

[NFC] Restore handling for deprecations in phpunit test runs

seamuslee001 committed:
Merge pull request #23990 from civicrm/5.52

5.52

eileenmcnaughton committed:
Merge pull request #23984 from seamuslee001/ref_full_package_upgrades_php8_1

[REF][PHP8.1] Upgrade oauth2-client,zetacomponents/base,htmlpurfier packages …

seamuslee001 committed:
Merge pull request #23991 from totten/5.52-civitest

Civi\Test - Fix leak in active module-list for headless test

eileenmcnaughton committed:
Merge pull request #23992 from civicrm/5.52

5.52

colemanw committed:
SearchKit - Fix adding deep joins

This fixes a UI bug that was preventing joins past a depth of 2 to be added.

colemanw committed:
Allow custom fields to have custom fields

Theoretically a custom field could have custom fields! Trippy...
Fixes dev/core#3731

seamuslee001 committed:
[REF][PHP8.1] Apply PR patch to fix issue in php8.1 where mysqli error reporting changed

seamuslee001 committed:
[REF][PHP8.1] Fix issues in core extensions where passing in NULL values is no longer supported

MegaphoneJon committed:
net_smtp upgrade to 1.10.0

seamuslee001 committed:
[REF][PHP8.1] A third batch of fixes where passing in NULL values causes deprecation notices in php8.1

seamuslee001 committed:
[REF][PHP8.1] Fix Return type hints and passing in NULL values causing deprecation notices in php8.1 in Property Bag

seamuslee001 committed:
Merge pull request #23998 from MegaphoneJon/net-smtp-110

net_smtp upgrade to 1.10.0

seamuslee001 committed:
Merge pull request #23993 from colemanw/searchKitJoinFix

SearchKit - Fix adding deep joins

mattwire committed:
Merge pull request #23974 from colemanw/filterFieldsAfform

Afform - Allow groups and tags and age to be filters

totten committed:
Merge pull request #23997 from seamuslee001/php8_1_null_ext

[REF][PHP8.1] Fix issues in core extensions where passing in NULL val…

totten committed:
Merge pull request #24000 from seamuslee001/property_bag_fixs

[REF][PHP8.1] Fix Return type hints and passing in NULL values causin…

eileenmcnaughton committed:
Merge pull request #23995 from colemanw/custom_custom

Allow custom fields to have custom fields

demeritcowboy committed:
Merge pull request #23999 from seamuslee001/null_part3_php81

[REF][PHP8.1] A third batch of fixes where passing in NULL values cau…

demeritcowboy committed:
fix missing upload

demeritcowboy committed:
Fix test. Don't hide errors.

colemanw committed:
Afform - provide easy way to add navigation menu from the form

seamuslee001 committed:
Merge pull request #24007 from demeritcowboy/pdf-error

Fix test. Don't hide errors. (Followup to 23999)

eileenmcnaughton committed:
Merge pull request #23996 from seamuslee001/php8_1_pear_db

[REF][PHP8.1] Apply PR patch to fix issue in php8.1 where mysqli erro…

eileenmcnaughton committed:
Merge pull request #24004 from demeritcowboy/upload-woes

dev/core#3736 - Fix uploaded file on message template form

seamuslee001 committed:
Merge pull request #23954 from eileenmcnaughton/poly

Poly-fill-me-up

demeritcowboy committed:
hash out of date

yashodha committed:
add classed to tpl

colemanw committed:
APIv4 - Flush navigation caches after writing or deleting navigation item

Historically the expectation was that this cache gets flushed manually.
I've kept that unchanged for legacy code (v3 create/delete and direct BAO::add)
but with APIv4 and onward the cache will get flushed automatically after a write operation.

colemanw committed:
Afform - Insta-refresh menubar when adding a navigation item

demeritcowboy committed:
Merge pull request #24012 from yashodha/add_Classes

add classes to the contribution view page

colemanw committed:
Merge pull request #24011 from demeritcowboy/composerlock

composer.lock hash out of date

demeritcowboy committed:
update to fix delete-and-reinstall problem

colemanw committed:
Afform - Post-submit redirect tokens

demeritcowboy committed:
lower minimum polyfill version

seamuslee001 committed:
[REF][PHP8.1] Batch 4 of fixing issues found in unit tests where passing NULL values triggers deprecations

Apply fixes as per Demeritcowboy

seamuslee001 committed:
Merge pull request #24015 from demeritcowboy/composer-patches

Update cweagans/composer-patches to 1.7.2 to fix delete-and-reinstall problem

seamuslee001 committed:
Merge pull request #24017 from demeritcowboy/polyfill-too-high

Lower minimum polyfill version since conflicts with drupal/core-recommended

eileenmcnaughton committed:
Merge pull request #24013 from colemanw/afformNav

Afform - provide easy way to add navigation menu from the form

mattwire committed:
Use fontawesome icon helper for activity/Page/AJAX

demeritcowboy committed:
Merge pull request #24021 from mattwire/activitypageajaxicons

Use fontawesome icon helper for activity/Page/AJAX

colemanw committed:
SearchKit - Allow html in columns

This adds a column type 'html' and defaults rich-text fields to use that column type.
Fixes dev/core#3723

MegaphoneJon committed:
add afform blocks to print summary

demeritcowboy committed:
fix crash removing an entry from a batch

colemanw committed:
Merge pull request #24024 from MegaphoneJon/print-summary-afform

Add afform blocks to print summary

totten committed:
Status Check - Report the overall status (accurately)

Overview
--------

The page-footer includes a summary message about the system status (eg "System
Status: Ok" or "System Status: Error").  The footer has inaccurately summarized
the status.

Before
------

Suppose you have 5 "Error" messages and 1 "OK" message. It will summarize as "System Status: Ok".

(It emphasizes the *least severe* status-message.)

Code is more complex.

After
------

Suppose you have 5 "Error" messages and 1 "OK" message. It will summarize as "System Status: Error".

(It emphasizes the *most severe* status-message.)

Code is less complex.

Comments
--------

The overall value is determined in `CRM_Utils_Check::checkAll()`.  Heretofore,
`checkAll()` sorted the messages by severity and then emphasized the *first*
visible message.

The problem is that the sort-order has flip-flopped, so "first" means different things:

* In v4.7, the messages were descending. So "first" was "most severe".
* In v5.39 (11d593edde82e7de962ec9056cfebe87975fb499), the order was ascending.
  So "first" was "least severe". (It appears accidental.)
* In v5.45 (728c1cc31d0d323f74f7b980a1e183fb499d4e9e), the order flipped back
  to descending, but *only* for the Angular (`civicrm/a/#/status`) -- not the footer.

IMHO, the flip-floppiness means that callers cannot rely on the order returned
by `checkAll()` -- they should instead do a higher-level sort (as in v5.45's
728c1cc31d0d323f74f7b980a1e183fb499d4e9e).  The `uasort()` within `checkAll()`
is therefore redundant.  We might as well skip it -- and simplify the logic.

yashodha committed:
remove pledge status as mandatory in Pledge report

totten committed:
Upgrader - Pick initial dispatch policy based on how far we've progressed in upgrade

totten committed:
CRM_Queue_Task - Fire internal events

totten committed:
(REF) CiviEventDispacher - Enable getDispatchPolicy()

totten committed:
Upgrader - Re-assess dispatch policy at the start of every upgrade task

The practical effect should be this:

* All tasks before `doCoreFinish()` run with the `upgrade.main` policy
* All tasks after `doCoreFinish()` run with the `upgrade.finish` policy

Framing this change is a bit delicate because tasks can run in different
ways (eg AJAX queue runner; headless queue-runner; cv queue-runner).  This
should have the same consequence in all runners.

mattwire committed:
Merge pull request #24023 from colemanw/searchKitHtml

SearchKit - Allow html in columns

demeritcowboy committed:
Merge pull request #24028 from yashodha/reports_improvements

remove pledge status as mandatory in Pledge report

totten committed:
Update CRM/Utils/Check.php

Co-authored-by: demeritcowboy <demeritcowboy@hotmail.com>

totten committed:
Merge pull request #23767 from colemanw/afformRequireAuthx

Afform - Require Authx

demeritcowboy committed:
Merge pull request #24018 from seamuslee001/php8_1_null_batch_4

[REF][PHP8.1] Batch 4 of fixing issues found in unit tests where pass…

yashodha committed:
fix 'Pledge Made Date' to show date and not time

eileenmcnaughton committed:
Merge pull request #24025 from demeritcowboy/batch-remove-2

dev/core#3050 - (Alternate) Fix crash removing an entry from a batch

demeritcowboy committed:
Merge pull request #24031 from yashodha/reports_improvements

Fix 'Pledge Made Date' display to show date and not time

demeritcowboy committed:
Merge pull request #24027 from totten/master552-status-footer-skipsort

Status Check - Report the overall status (accurately)

civicrm-builder commented:
Can one of the admins verify this patch?

colemanw committed:
Fix ActionLog BAO to not overwrite existing data

demeritcowboy commented:
Hi,
You'll need to make the base branch the master branch. See https://docs.civicrm.org/dev/en/latest/tools/git. You may also want to take a look at https://docs.civicrm.org/dev/en/latest/core/contributing.

In terms of how this feature would work, maybe best discussed in a lab ticket at https://lab.civicrm.org instead of here but another option besides profiles would be as one of the dropdown actions on Find Cases results, since you may be able to borrow code from the existing action for _contact_ search results (add to case as role).

seamuslee001 committed:
Merge pull request #24035 from colemanw/actionLogFix

Fix ActionLog BAO to not overwrite existing data

seamuslee001 committed:
Merge pull request #23895 from colemanw/searchKitManaged

SearchKit - Switch to APIv4 for managed entities

seamuslee001 committed:
Merge pull request #23970 from MegaphoneJon/false-not-zero

APIv4 - Fix mishandling of boolean custom values

colemanw committed:
Merge pull request #24016 from colemanw/afformTokens

Afform - Post-submit redirect tokens

colemanw committed:
Merge pull request #24030 from totten/master-run-ext-upgrades

dev/core#3490 - Apply extension updates after core updates

totten committed:
Merge pull request #23896 from colemanw/afformManaged

Afform - Switch to APIv4 for managed afform_type option group

totten committed:
Merge pull request #24038 from seamuslee001/5.52

[REF] Upgrade DOMPDF to 2.x branch

seamuslee001 committed:
Merge branch '5.52'

seamuslee001 committed:
Merge pull request #24042 from seamuslee001/master

5.52

totten committed:
(NFC) CRM_Core_Permission - More docblocks

totten committed:
composer.json - Update scssphp and related libraries (for PHP 8.1)

Overview
--------

Update scssphp/scssphp, civicrm/composer-compile-plugin, and civicrm/composer-compile-lib.

Before
-------

Large number of warnings are generated in scssphp / php81, leading to an overflow

```
Compiling additional files (For full details, use verbose "-v" mode.)
Compile: Generate CCL wrapper functions
Compile: Greenwich CSS (dist/bootstrap3.css)

Fatal error: Allowed memory size of 1610612736 bytes exhausted (tried to allocate 36864 bytes) in phar:///Users/totten/bknix/bin/composer/vendor/symfony/process/Pipes/UnixPipes.php on line 132

Check https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors for more info on how to handle out of memory errors.Subcommand @composer compile  returned with error code 255

In ComposerPassthru.php line 72:

  Subcommand @composer compile  returned with error code 255

install [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--dev] [--no-suggest] [--no-dev] [--no-autoloader] [--no-progress] [--no-install] [--audit] [--audit-format AUDIT-FORMAT] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--] [<packages>...]
```

After
-----

Compilation works

Comments
--------

Generated via: `composer update civicrm/composer-compile-lib scssphp/scssphp civicrm/composer-compile-plugin`

demeritcowboy committed:
Merge pull request #24044 from totten/master-scss

composer.json - Update scssphp and related libraries (for PHP 8.1)

demeritcowboy committed:
Merge pull request #24043 from totten/master-nfc-perm

(NFC) CRM_Core_Permission - More docblocks

demeritcowboy commented:
Hi @khennhofer can you start by making a lab ticket at https://lab.civicrm.org? I'm going to close this since opening a new PR against the master branch will be easier than trying to rebase this and it's not possible to see here what the actual change is in this PR. See also the links above for docs for contributing.
